Common/cpp/Tools/JSISerializer.cpp

```diff
@@ -0,0 +1,231 @@
+#include "JSISerializer.h"
+#include <cxxabi.h>
+
+bool checkJSCollectionType(
+    jsi::Runtime &rt,
+    const jsi::Object &obj,
+    std::string expectedType) {
+  const jsi::Function &getPrototype =
+      rt.global()
+          .getPropertyAsObject(rt, "Object")
+          .getPropertyAsFunction(rt, "getPrototypeOf");
+
+  std::string result = getPrototype.call(rt, obj).toString(rt).utf8(rt);
+  std::string pattern = "[object " + expectedType + "]";
+
+  return result == pattern;
+}
```

Comment:
wrap with unnamed namespace in order to make this function locally available only in this file

```suggestion
namespace {
bool checkJSCollectionType(
    jsi::Runtime &rt,
    const jsi::Object &obj,
    std::string expectedType) {
  const jsi::Function &getPrototype =
      rt.global()
          .getPropertyAsObject(rt, "Object")
          .getPropertyAsFunction(rt, "getPrototypeOf");

  std::string result = getPrototype.call(rt, obj).toString(rt).utf8(rt);
  std::string pattern = "[object " + expectedType + "]";

  return result == pattern;
}
}
```